docs/05_operations/08_update-queue/04_jms.mdx

```diff
@@ -0,0 +1,133 @@
+
+### Basic configuration
+
+To use Jakarta Messaging in your Genesis application, you need to set the `MqLayer` in the system definition.
+
+```kotlin {title="genesis-system-definition.kts"}
+systemDefinition {
+    global {
+        ...
+        item(name = "MqLayer", value = "JMS")
+        ...
+    }
+}
+```
+
+All Available Config Variables:
+
+| Config Item                      | Description                          | Default                  |
+|----------------------------------|--------------------------------------|--------------------------|
+| `JmsHost` (required)             | The host name, can hold multiple values separated by comma, allowing HA failover | `http://localhost:61616` |
+| `JmsUsername`                    | Username to connect | |
+| `JmsPassword`                    | Password to connect (can be encrypted using `GenesisKey`) | |
+| `JmsTopicSubscriptionMode`       | Enum value (TRANSIENT, PERSISTENT) | `TRANSIENT`|
+| `JmsMaxConnectionRetryOnStartup` | The number of connections the process will attempt to make to the broker, before declaring the process `DOWN` | 20 |
+| `JmsReconnectDelayMs`            | The amount of time in milliseconds (exponentially) for a connection reattempt to be made, in case connectivity or fault happens | 1000 |
+| `JmsRetryToSendDelayMs`          | The amount of time in milliseconds for retrying sending a message in case of failure. | 1000 |
+
+### Durable consumers
+
+It is possible to configure certain process to receive messages that were sent when the process was offline. This might give another way for processes to recover from missing messages without the need of replaying an entire data stream.
+
+For example, assume you have an application named `alpha` and you are going to setup `ALPHA_CONSOLIDATOR` to be a durable consumer of the update queue.
+
+1. Create a configuration file for the `ALPHA_CONSOLIDATOR` process with the contents:
+
+```kotlin {title="alpha-consolidator-process-config.kts"}
+process {
+    systemDefinition {
+        item(name = "JmsTopicSubscriptionMode", value = "PERSISTENT")
+    }
+}
+```
+
+2. Associate the newly created configuration:
+
+```xml {title="process.xml"} {8}
+<process name="ALPHA_CONSOLIDATOR">
+    <groupId>ALPHA</groupId>
+    <start>true</start>
+    <options>-Xmx256m -DRedirectStreamsToLog=true -DXSD_VALIDATE=false -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5106</options>
+    <module>genesis-pal-consolidator</module>
+    <package>global.genesis.pal.consolidator</package>
+    <script>alpha-consolidator.kts</script>
+    <config>alpha-consolidator-process-config.kts</config>
+    <language>pal</language>
+    <primaryOnly>true</primaryOnly>
+</process>
+```
+
+3. Perform `genesisInstall` to apply the configuration and start processes normally.
+
+:::danger
+Do not define `JmsTopicSubscriptionMode` at Global level as this will cause all messages to be stored and delivered once process are back online.
```

Comment:
Wording here: need help to emphasize this is really BAD, and must not be done (I wish we could have an easier way to prevent but we don't).